backend.py

```diff
@@ -207,6 +223,53 @@ def get_start_state(map_name):
     return state
 
 
+def check_wall(coordinates, direction, state):
+    """
+    Check wall in the direction of the move.
+
+    coordinates: tuple of x and y coordinate
+    direction: object of Direction class
+    state: object of State class
+
+    Return a boolean.
+
+    True - There isn't wall, robot can move.
+    False - There is wall, robot can't move.
+    """
+    old_tiles = state.get_tiles(coordinates)
+    # On the current tile: Check wall in the direction of next move.
+    for tile in old_tiles:
+        move_from = tile.can_move_from(direction)
+        if move_from is False:
+            # On the current tile: There is a wall in the direction of the move.
+            return False
+    if move_from:
+        # There is no wall, so get new coordinates.
+        next_coordinates = get_next_coordinates(coordinates, direction)
+        # Get new list of tiles
+        new_tiles = state.get_tiles(next_coordinates)
+        # Check wall on the next tile in the direction of the move.
+        for tile in new_tiles:
+            move_to = tile.can_move_to(direction)
+            if move_to is False:
+                # On the next tile: There is a wall in the direction
+                # of the move.
+                return False
+        if move_to:
+            return True
+
+
+def get_next_coordinates(coordinates, direction):
```

Comment:
Je to pomocná metoda/funkce, ale pořád je to metoda, kterou můžeme aplikovat pouze na robota. Je ok, když je vytažená mimo robotí třídu? Jaká je v této otázce konvence?

Reply:
No ona to právě jenom robotí metoda není. Stejný kus kódu, který tato metoda obsahuje, je třeba i v metodě `shoot_robot` v podtřídě `LaserTile` a k robotovi se tam nevztahuje.

Reply:
Aha! Tak to pardon, čtu nepořádně.